picard/file.py

```diff
@@ -780,20 +776,26 @@ def _info(self, metadata, file):
             metadata['~format'] = self.NAME
         else:
             metadata['~format'] = self.__class__.__name__.replace('File', '')
-        self._add_path_to_metadata(metadata)
+        self._update_filesystem_metadata(metadata)
 
-    def _add_path_to_metadata(self, metadata):
+    def _update_filesystem_metadata(self, metadata):
         metadata['~dirname'] = os.path.dirname(self.filename)
-        filename, extension = os.path.splitext(os.path.basename(self.filename))
-        metadata['~filename'] = filename
+        filename_no_ext, extension = os.path.splitext(os.path.basename(self.filename))
+        metadata['~filename'] = filename_no_ext
         metadata['~extension'] = extension.lower()[1:]
 
+        filename_encoded = encode_filename(self.filename)
+        try:
+            metadata['~filesize'] = os.path.getsize(filename_encoded)
+        except OSError as ex:
```

Comment:
As said in a previous comment, I think we can regroup this with ctime/mtime below and use the same try/except block since they all raise OSError.

Reply:
Ok, I combined it